src2/api/index.js

```diff
@@ -0,0 +1,78 @@
+import 'isomorphic-fetch'
+import _ from 'underscore'
+import Promise from 'bluebird'
+import querystring from 'querystring'
+import url from 'url'
+
+class Api {
+  var cache = {};
+
+  get(url) {
+    if (this.cache[url]) {
+      return Promise.resolve(this.cache[url]);
+    } else {
+      return fetch(url).then(function(response) {
+        return response.text()
+      }).then(function(text) {
+        return this.cache[url] = text
+      });
+    }
+  }
+
+  getJson(url) {
+    return this.get(url).then(JSON.parse)
+  }
+
+  flush() {
+    this.cache = {}
+  }
+
+  buildUrl(endpoint, cube, path, params) {
+    params = params || {};
+    var api = endpoint;
+    api = endpoint[api.length - 1] == '/' ? api.slice(0, api.length - 1) : api;
+    api = api + '/cubes/' + cube + '/' + path;
```

Comment:
@borysyuk and cc @kravets-levko seeing as we are using ES6, try the new template strings. Example:

```
let api = `${api}/cubes/${cube}/path`
```
Reference: https://github.com/okfn/spend-publishing-dashboard/blob/master/app/utils.js#L19